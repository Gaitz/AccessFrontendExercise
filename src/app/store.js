import { configureStore } from "@reduxjs/toolkit";
import userListSlice from "../features/list/userListSlice";
import userDetailSlice from "../features/detail/userDetailSlice";

const store = configureStore({
  reducer: { list: userListSlice, detail: userDetailSlice },
});

export default store;
