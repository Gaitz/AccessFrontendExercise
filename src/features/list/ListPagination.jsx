import React from "react";
import styles from "./ListPagination.module.css";
import { useDispatch, useSelector } from "react-redux";
import { appendPageIndex, goToPageIndex } from "./userListSlice";

const ListPagination = () => {
  const dispatch = useDispatch();

  const pageIndex = useSelector((state) => state.list.pageIndex);
  const currentPage = pageIndex + 1;

  return (
    <div className={styles.paginatedBlock}>
      <p>
        Page Index:{" "}
        {Array(currentPage)
          .fill(currentPage)
          .map((v, i) => {
            const isDisabled = i === pageIndex;
            return (
              <a
                key={i}
                href={`/page/${i}`}
                onClick={(event) => {
                  event.preventDefault();
                  dispatch(goToPageIndex(i));
                }}
                className={`${styles.pageIndexLink} ${
                  isDisabled ? styles.disabledLink : ""
                }`}
              >
                {i}
              </a>
            );
          })}
      </p>
      <button
        onClick={(event) => {
          event.preventDefault();
          dispatch(appendPageIndex(1));
        }}
      >
        Get Next Page
      </button>
    </div>
  );
};

export default ListPagination;
