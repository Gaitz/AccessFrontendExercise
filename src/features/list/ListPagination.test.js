import React from "react";
import { Provider } from "react-redux";
import store from "../../app/store";
import {
  fireEvent,
  getByText,
  queryByText,
  render,
} from "@testing-library/react";
import ListPagination from "./ListPagination";

describe("ListPagination component test", () => {
  test("snapshot ListPagination", () => {
    const { container } = render(
      <Provider store={store}>
        <ListPagination></ListPagination>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  test("Get Next Page", () => {
    const { container, getByText } = render(
      <Provider store={store}>
        <ListPagination></ListPagination>
      </Provider>
    );
    const nextPageButton = getByText("Get Next Page");
    fireEvent.click(nextPageButton);
    expect(container).toMatchSnapshot();
  });

  describe("page index link", () => {
    let container, getByText, queryByText;

    beforeEach(() => {
      ({ container, getByText, queryByText } = render(
        <Provider store={store}>
          <ListPagination></ListPagination>
        </Provider>
      ));
    });

    test("by click disabled link", () => {
      const initialPageIndex = getByText("0");
      fireEvent.click(initialPageIndex);
      expect(container).toMatchSnapshot();
    });

    test("by click active link", () => {
      const nextPageButton = getByText("Get Next Page");

      fireEvent.click(nextPageButton);
      fireEvent.click(nextPageButton);
      const secondPageIndex = queryByText("1");
      fireEvent.click(secondPageIndex);

      expect(container).toMatchSnapshot();
    });
  });
});
