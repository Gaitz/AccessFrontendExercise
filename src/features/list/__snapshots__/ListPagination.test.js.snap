// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPagination component test Get Next Page 1`] = `
<div>
  <div
    class="paginatedBlock"
  >
    <p>
      Page Index:
       
      <a
        class="pageIndexLink "
        href="/page/0"
      >
        0
      </a>
      <a
        class="pageIndexLink disabledLink"
        href="/page/1"
      >
        1
      </a>
    </p>
    <button>
      Get Next Page
    </button>
  </div>
</div>
`;

exports[`ListPagination component test page index link by click active link 1`] = `
<div>
  <div
    class="paginatedBlock"
  >
    <p>
      Page Index:
       
      <a
        class="pageIndexLink "
        href="/page/0"
      >
        0
      </a>
      <a
        class="pageIndexLink disabledLink"
        href="/page/1"
      >
        1
      </a>
    </p>
    <button>
      Get Next Page
    </button>
  </div>
</div>
`;

exports[`ListPagination component test page index link by click disabled link 1`] = `
<div>
  <div
    class="paginatedBlock"
  >
    <p>
      Page Index:
       
      <a
        class="pageIndexLink disabledLink"
        href="/page/0"
      >
        0
      </a>
    </p>
    <button>
      Get Next Page
    </button>
  </div>
</div>
`;

exports[`ListPagination component test snapshot ListPagination 1`] = `
<div>
  <div
    class="paginatedBlock"
  >
    <p>
      Page Index:
       
      <a
        class="pageIndexLink disabledLink"
        href="/page/0"
      >
        0
      </a>
    </p>
    <button>
      Get Next Page
    </button>
  </div>
</div>
`;
